exasol/analytics/audit/audit.py

```diff
@@ -23,3 +38,113 @@ def __init__(
             name=TableNameImpl(table_name, SchemaName(db_schema)),
             columns=(BaseAuditColumns.all + additional_columns),
         )
+        self._column_names = [c.name for c in self.columns]
+
+    def augment(self, queries: Iterator[Query]) -> Iterator[str]:
+        for query in queries:
+            if not query.audit:
+                yield query.query_string
+            elif isinstance(query, AuditQuery):
+                yield self._insert(query)
+            elif isinstance(query, ModifyQuery):
+                yield from self._wrap(query)
+            else:
+                raise TypeError(
+                    f"Unexpected type {type(query)}" f" of query {query.query_string}"
+                )
+
+    def _insert(self, query: AuditQuery) -> str:
+        sub_query_alias = TableNameImpl("SUB_QUERY")
+        subquery_columns = {
+            c.name.name: ColumnName(c.name.name, sub_query_alias)
+            for c in output_columns(query.select_with_columns)
```

Comment:
What if you don't have a subquery

Reply:
This is what `output_columns()` is for.

Reply:
Test is added and works as expected:
https://github.com/exasol/advanced-analytics-framework/pull/269/files#diff-bdb3637071e80278ae5dee916c298cf8ab7c1c47de7101a0f160cc1e36bf7a57R49

Reply:
See latest push - and thanks for your proposal making the implementation shorter, simpler, more elegant.